Tasks/Src/RTOS_BPS.c

```diff
@@ -52,6 +52,15 @@ void RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options) {
 void RTOS_BPS_TaskCreate(void){
 
 }
-void RTOS_BPS_TimeDelay(void){
 
-}
\ No newline at end of file
+/**
+ * @brief: Creates a Time Delay.
+ * @param dly Defines how many ticks to delay for.
+ * @return none
+ */
+void RTOS_BPS_TimeDelay(BPS_OS_TICK dly, BPS_OS_OPT opt){
```

Comment:
Can you update the rest of the code to call this function instead of `OSTimeDly`? It would also help because then the diff would show which parameters we are passing to it. (I am not sure we are passing different parameters for opt)